src/sail_json_backend/json.ml

```diff
@@ -231,20 +232,63 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
 
+let rec filter_non_operands components =
+  match components with
+  | [] -> []
+  | hd :: tl when String.trim hd = "spc" -> tl
+  | _ -> filter_non_operands (List.tl components)
+
+let rec extract_operands k filtered_components =
+  match filtered_components with
+  | [] -> []
+  | hd :: tl ->
+      if Str.string_match (Str.regexp ".+(\\(.*\\))") hd 0 then (
+        let operand = Str.matched_group 1 hd in
+        let elements = Str.split (Str.regexp ",") operand in
```

Comment:
very minor and subjective, but "operand" isn't the best variable name here. Since it is only used in the next line, anyway, let's combine these two lines:
```
let elements = Str.split (Str.regexp ",") (Str.matched_group 1 hd) in
```

Reply:
Done : )